clustering/infinispan/client/spi/src/main/java/org/wildfly/clustering/infinispan/client/near/CaffeineNearCache.java

```diff
@@ -43,18 +43,23 @@ public CaffeineNearCache(Cache<K, MetadataValue<V>> cache) {
     }
 
     @Override
-    public void put(K key, MetadataValue<V> value) {
-        this.map.put(key, value);
+    public boolean putIfAbsent(K key, MetadataValue<V> value) {
+        return this.map.putIfAbsent(key, value) == null;
     }
 
     @Override
-    public void putIfAbsent(K key, MetadataValue<V> value) {
-        this.map.putIfAbsent(key, value);
+    public boolean remove(K key) {
+        return this.map.remove(key) != null;
     }
 
     @Override
-    public boolean remove(K key) {
-        return this.map.remove(key) != null;
+    public boolean remove(K key, MetadataValue<V> value) {
+        return this.map.replace(key, value) != null;
```

Comment:
Should this be `this.map.remove(key ,value)`?

Reply:
:facepalm: Another auto-completion victim...

Reply:
Fixed.  Good catch @jamezp .